typing/predef.ml

```diff
@@ -163,54 +236,86 @@ let mk_add_type add_type type_ident ?manifest
-    let decl =
-      {type_params = [param];
-       type_arity = 1;
-       type_kind = kind param;
-       type_loc = Location.none;
-       type_private = Asttypes.Public;
-       type_manifest = None;
-       type_variance = [variance];
-       type_separability = [separability];
-       type_is_newtype = false;
-       type_expansion_scope = lowest_level;
-       type_attributes = [];
-       type_immediate = Unknown;
-       type_unboxed_default = false;
-       type_uid = Uid.of_predef_id type_ident;
-      }
-    in
-    add_type type_ident decl env
-  and add_continuation type_ident env =
-    let tvar1 = newgenvar() in
-    let tvar2 = newgenvar() in
-    let arity = 2 in
-    let decl =
-      {type_params = [tvar1; tvar2];
-       type_arity = arity;
-       type_kind = Type_abstract Definition;
-       type_loc = Location.none;
-       type_private = Asttypes.Public;
-       type_manifest = None;
-       type_variance = [Variance.contravariant; Variance.covariant];
-       type_separability = Types.Separability.default_signature ~arity;
-       type_is_newtype = false;
-       type_expansion_scope = lowest_level;
-       type_attributes = [];
-       type_immediate = Unknown;
-       type_unboxed_default = false;
-       type_uid = Uid.of_predef_id type_ident;
-      }
-    in
-    add_type type_ident decl env
+    { (decl0 ~kind:(kind param) type_ident) with
+      type_params = [param];
+      type_arity = 1;
+      type_variance = [variance];
+      type_separability = [separability];
+    }
   in
+  let decl2
+      ~variance:(var1, var2)
+      ?separability:((sep1, sep2) = (Separability.Ind, Separability.Ind))
+      ?(kind = fun _ _ -> Type_abstract Definition)
+      type_ident
+    =
+    let param1, param2 = newgenvar (), newgenvar () in
+    { (decl0 ~kind:(kind param1 param2) type_ident) with
+      type_params = [param1; param2];
+      type_arity = 2;
+      type_variance = [var1; var2];
+      type_separability = [sep1; sep2];
+    }
+  in
+  let cstr id args =
+    {
+      cd_id = id;
+      cd_args = Cstr_tuple args;
+      cd_res = None;
+      cd_loc = Location.none;
+      cd_attributes = [];
+      cd_uid = Uid.of_predef_id id;
+    }
+  in
+  let variant constrs =
+    Type_variant (constrs, Variant_regular) in
+  function
+  | `Int -> decl0 ~immediate:Always ident_int
+  | `Char -> decl0 ~immediate:Always ident_int
+  | `String -> decl0 ident_string
+  | `Bytes -> decl0 ident_bytes
+  | `Float -> decl0 ident_float
+  | `Bool ->
+      let kind = variant [cstr ident_false [];
+                          cstr ident_true []] in
+      decl0 ~immediate:Always ~kind ident_bool
+  | `Unit ->
+      let kind = variant [cstr ident_void []] in
+      decl0 ~immediate:Always ~kind ident_unit
+  | `Exn -> decl0 ~kind:Type_open ident_exn
+  | `Eff ->
+      let kind _ = Type_open in
+      decl1 ~variance:Variance.full ~kind ident_eff
+  | `Continuation ->
+      let variance = Variance.(contravariant, contravariant) in
```

Comment:
This was set to `[Variance.contravariant; Variance.covariant]` before these changes (line 202).

Reply:
Oops! Thanks.